testutils/src/main/java/com/microsoft/identity/internal/testutils/MockHttpClient.java

```diff
@@ -0,0 +1,197 @@
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod, String)
+     */
+    public static void setHttpResponse(@NonNull final HttpResponse httpResponse) {
+        MockHttpClient.setHttpResponse(httpResponse, null, ".*");
+    }
+
+    /**
+     * A shorthand for intercepting http requests by giving the response for all outgoing
+     * requests with the specified http method
+     *
+     * @param httpResponse - the http response
+     * @param method       - the method that will be returning the response
+     * @see MockHttpClient#setHttpResponse(HttpResponse)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod, String)
+     */
+    public static void setHttpResponse(@NonNull final HttpResponse httpResponse, @NonNull HttpMethod method) {
+        MockHttpClient.setHttpResponse(httpResponse, method, ".*");
+    }
+
+    /**
+     * A shorthand for intercepting http requests by giving the response for all outgoing
+     * requests that match the specified url regex.
+     *
+     * @param httpResponse - the http response
+     * @param urlRegex     - the url regex to match requests
+     * @see MockHttpClient#setHttpResponse(HttpResponse)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod, String)
+     */
+    public static void setHttpResponse(@NonNull final HttpResponse httpResponse, @NonNull final String urlRegex) {
+        MockHttpClient.setHttpResponse(httpResponse, null, urlRegex);
+    }
+
+    /**
+     * A shorthand for intercepting requests by giving the response for all outgoing requests that
+     * match the specified url regex and http method
+     *
+     * @param httpResponse - the http response
+     * @param method       - the request method
+     * @param urlRegex     - the url regex pattern to match
+     * @see MockHttpClient#setHttpResponse(HttpResponse)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, String)
+     * @see MockHttpClient#setHttpResponse(HttpResponse, HttpMethod)
+     */
+    public static void setHttpResponse(@NonNull final HttpResponse httpResponse,
+                                       @Nullable final HttpMethod method,
+                                       @NonNull final String urlRegex) {
+        MockHttpClient.setInterceptor(new HttpRequestInterceptor() {
+            @Override
+            public HttpResponse intercept(@NonNull HttpMethod httpMethod,
+                                          @NonNull URL requestUrl,
+                                          @NonNull Map<String, String> requestHeaders,
+                                          @Nullable byte[] requestContent) {
+                return httpResponse;
+            }
+        }, method, urlRegex);
+    }
+
+    /**
+     * Clears the existing interceptors
+     */
+    public static void reset() {
```

Comment:
This will clear any and all the interceptors, set by any prevoously running code. do we want that.. 
Or we want to only clear the interceptors set by a particular test case/class. 
In that case we should let the caller instantiate MockHttpClient instead of providing static field for interceptor

Reply:
If we're building an entire mocking framework for Http, I would suggest that we stop and use one instead.

Reply:
And if we're doing interception and alteration in the mock client, I would suggest that these features are better part of an actual client.  And that we should add them to the real client class.

Reply:
+1 for usinge a mocikng framework (like mockito) to mock.

Reply:
discussed offline with Shahzaib, looks like we have faced some problems in using mockito in the past, and decided to use shadows instead

Reply:
Yes, I faced issues with Mockito as well.